### PR TITLE
numpy_for_reshape

### PR DESCRIPTION
**Pull request by ben-bou**
## Description

Use numpy to check the shape in reshape. The way it currently is in #796 works, this does the same but uses numpy. Therefore allows more numpy consistency (e.g. see the changed test) and simplifies `try ... except` 

Issue/s resolved: #

## Changes proposed:
- Use numpy for checking and sanitation of shape
-
-
-

## Due Diligence

- [ ] All split configurations tested
- [ ] Multiple dtypes tested in relevant functions
- [ ] Documentation updated (if needed)
- [ ] Updated changelog.md under the title "Pending Additions"

#### Does this change modify the behaviour of other functions? If so, which?
yes / no

### CODE REVIEW

**Comment by mtar** [comment]
GPU cluster tests are currently disabled on this Pull Request.

**Comment by ClaudiaComito** [comment]
run tests

**Comment by ClaudiaComito** [comment]
ok to test

**Comment by ClaudiaComito** [comment]
ok to test

File: heat/core/manipulations.py [hunk 1]
```diff
@@ -1724,39 +1724,27 @@ def reshape(a: DNDarray, *shape: Union[int, Tuple[int, ...]], **kwargs) -> DNDar
...
-            if a.size % shape_size != 0:
-                raise ValueError(
-                    "cannot reshape array of size {} into shape {}".format(a.size, shape)
-                )
-        else:
-            if a.size != shape_size:
-                raise ValueError(
-                    "cannot reshape array of size {} into shape {}".format(a.size, shape)
-                )
+    shape = np_proxy.reshape(shape).shape  # sanitized shape according to numpy
```
Comment (ClaudiaComito) [inline heat/core/manipulations.py:1745]: I like this. It also makes lines 1790-1796 redundant, right?

**Review by ClaudiaComito — APPROVED** [review]
Thanks a lot @ben-bou !

**Pull request merged**